### PR TITLE
🌱 add latency metrics for resource resync.

### PR DESCRIPTION
**Pull request by morvencao**
## Summary

This PR adds metrics to measure the duration of resource spec resync and status resync operations.

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/cloudevents/generic/metrics_collector.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	// Register the metrics for resource resync:
+	RegisterResourceResyncMetrics()
```
Comment (skeeey) [inline pkg/cloudevents/generic/metrics_collector.go:11]: we may only expose the `resourceSpecResyncDurationMetric` and `resourceStatusResyncDurationMetric`, and let sdk user to select register them or not?

And expecting the resync metrics, do we need to add other metrics in sdk level? e.g. the pub/sub?

Comment (morvencao) [reply pkg/cloudevents/generic/metrics_collector.go]: 1. removed `init` function to let sdk user to register metrics.
2. for other metrics, we can start by covering some in the Maestro gRPC server implementation. Additional metrics can be added later based on user feedback. comments @clyang82 ?

Comment (clyang82) [reply pkg/cloudevents/generic/metrics_collector.go]: agree. let us wait for feedback before emitting more metrics.

File: pkg/cloudevents/generic/metrics_collector.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+// metricsLabels - Array of labels added to metrics:
+var metricsLabels = []string{
+	metricsSourceLabel,
+	metricsClusterLabel,
+	metrucsDataTypeLabel,
+}
+
+// Names of the metrics:
```
Comment (qiujian16) [inline pkg/cloudevents/generic/metrics_collector.go:26]: we need more docs here on what the format of metrics and what is the expected output as an example

File: pkg/cloudevents/generic/metrics_collector.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+const (
+	specResyncDurationMetric   = "spec_resync_duration_seconds"
+	statusResyncDurationMetric = "status_resync_duration_seconds"
+)
+
+// Description of the resource spec resync duration metric:
+var resourceSpecResyncDurationMetric = prometheus.NewHistogramVec(
```
Comment (qiujian16) [inline pkg/cloudevents/generic/metrics_collector.go:33]: can we have some tests?

File: pkg/cloudevents/generic/metrics_collector.go (hunk 2 above)
Comment (morvencao) [reply pkg/cloudevents/generic/metrics_collector.go:26]: added some comments and sample metrics expected collected.

File: pkg/cloudevents/generic/metrics_collector.go (hunk 3 above)
Comment (morvencao) [reply pkg/cloudevents/generic/metrics_collector.go:48]: added a simple test to check the metrics label and sum/count are as expected.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**